cpp/include/cudf/io/orc.hpp

```diff
@@ -51,8 +51,8 @@ class orc_reader_options_builder;
 class orc_reader_options {
   source_info _source;
 
-  // Names of column to read; empty is all
-  std::vector<std::string> _columns;
+  // Names of column to read; `nullptr` is all
+  std::optional<std::vector<std::string>> _columns;
```

Comment:
I suspect that this will again cause the error `optional is undefined`. Please include the `optional` header.